h2o-py/h2o/explanation/_explain.py

```diff
@@ -951,33 +951,100 @@ def mapping(x):
+    grouping_variable_type = "string" if type(grouping_variable_value) is str else "double" # todo test this
+    bin_type = data_to_append.col_types[0]
+
+    col_header = ["sample_id", "row_id", "column", "mean_response", "simulated_x_value", "is_original_observation"]
+    col_types = ["string", "int", "string", response_type, bin_type, "bool"]
+    table_header = "ICE plot graphing output" + kwargs.get("group_label", "")
+
+    if grouping_variable_value is not None:
+        col_header.append("grouping_variable_value")
+        col_types.append(grouping_variable_type)
+    centering_value = None
+    if centered:
+        col_header.append("centered_response")
+        col_types.append(response_type)
+        centering_value = data_to_append['mean_response'][0]
+    if show_logoods:
+        col_header.append("log(odds)")
+        col_types.append(response_type)
+
+    if graphing_data is None:
+        return h2o.two_dim_table.H2OTwoDimTable(col_header=col_header, col_types=col_types, table_header=table_header,
+                                                cell_values=_extract_graphing_data_values(data_to_append, frame_id,
+                                                                                          grouping_variable_value,
+                                                                                          original_observation_value,
+                                                                                          centering_value, show_logoods,
+                                                                                          row_id))
+    if data_to_append is None:
+        return graphing_data
+
+    new_values = graphing_data._cell_values + _extract_graphing_data_values(data_to_append, frame_id,
+                                                                            grouping_variable_value,
+                                                                            original_observation_value, centering_value,
+                                                                            show_logoods, row_id)
+    return h2o.two_dim_table.H2OTwoDimTable(col_header=graphing_data.col_header, col_types=graphing_data.col_types,
+                                            cell_values=new_values, table_header=table_header)
+
+
+def _extract_graphing_data_values(data, frame_id, grouping_variable_value, original_observation, centering_value,
+                                  show_logodds, row_id):
+    res_data = []
+    column = data.col_header[0]
+    for row in data.cell_values:
+        new_row = [frame_id, row_id, column, row[1], row[0], original_observation == row[0]]
+        if grouping_variable_value is not None:
+            new_row.append(grouping_variable_value)
+        if centering_value is not None:
+            new_row.append(row[1] - centering_value)
+        if show_logodds:
+            new_row.append(np.log(row[1] / (1 - row[1])))
```

Comment:
QQ: Is it actually possible, that the response type is something else than the numeric type, and this fails? Also is handling of sth like `Inf` and `NaN` handled somewhere? If the output is somehow serialized e.g. with JSON, this could lead to unexpected issues.

Reply:
I've checked and `mean_response` (`row[1]`) is always numeric/double. In output table, everytime when `centered_response` column or `log(odds)` column is present, also `mean_response` column is present. So, if it happenes that `mean_response` value is `Inf` or 'nan', the results in `centered_response`/ `log(odds)`  can be `inf` or `nan` as a result of inf/nan arithmetics but it will be directly visible why.  And regarding serialization... H2OTwoDimTable is a standard way how we output various numbers in 2D tables and can be easily converted to pandas DataFrame like `table.as_data_frame()` and that can be saved to csv easily, so I think this is more common way how to deal with the output than serialize it with JSON.. Or are you pointing to something else?